core/node/api_server/src/web3/namespaces/debug.rs

```diff
@@ -71,6 +104,66 @@ impl DebugNamespace {
         }
     }
 
+    fn map_flatten_call(
+        call: Call,
+        calls: &mut Vec<DebugCallFlat>,
+        trace_address: &mut Vec<usize>,
+        only_top_call: bool,
+        transaction_position: usize,
+        transaction_hash: H256,
+    ) {
+        let subtraces = call.calls.len();
+        let debug_type = match call.r#type {
+            CallType::Call(_) => DebugCallType::Call,
+            CallType::Create => DebugCallType::Create,
+            CallType::NearCall => unreachable!("We have to filter our near calls before"),
+        };
+
+        let result = if call.error.is_none() {
+            Some(CallResult {
+                output: web3::Bytes::from(call.output),
+                gas_used: U256::from(call.gas_used),
+            })
+        } else {
+            None
+        };
+
+        calls.push(DebugCallFlat {
+            action: Action {
+                call_type: debug_type,
+                from: call.from,
+                to: call.to,
+                gas: U256::from(call.gas),
+                value: call.value,
+                input: web3::Bytes::from(call.input),
+            },
+            result,
+            subtraces,
+            traceaddress: trace_address.clone(), // Clone the current trace address
+            transaction_position,
+            transaction_hash,
+            r#type: DebugCallType::Call,
+        });
+
+        if !only_top_call {
+            call.calls
+                .into_iter()
+                .enumerate()
+                .for_each(|(number, call)| {
```

Comment:
Nit: `for` cycle is arguably more idiomatic, and would be easier to read. Yes, I understand it's copy-pasted code, but still.